### ISSUE TITLE
Feature to start AWS instances but not Scylla nodes

### ISSUE THREAD

**Issue by kbr-** [scylladb/scylla-cluster-tests#2282]
Bootstrapping a cluster in SCT is extremely slow. It could be vastly improved if we had the ability to start just the AWS instances -- which we could do all at once in parallel -- and only then bootstrap a Scylla node on each instance sequentially.

**Comment by fruch** [comment]
You just described what SCT is doing today...

**Comment by kbr-** [comment]
Not as much as it could.
Here's an example output.log from https://github.com/scylladb/scylla-cluster-tests/pull/2283: 
[output.log](https://github.com/scylladb/scylla-cluster-tests/files/4812466/output.log)

Notice these lines:
```
< t:2020-06-19 18:48:43,438 f:cluster.py      l:2647 c:sdcm.cluster         p:INFO  > Cluster multi-cluster-kbraun-db-cluster-cd32b68f (AMI: ['ami-007f189e38720579f'] Type: i3.large): Init nodes
(...)
< t:2020-06-19 18:50:22,808 f:cluster.py      l:2647 c:sdcm.cluster         p:INFO  > Cluster multi-cluster-kbraun-oracle-db-cluster-cd32b68f (AMI: ['ami-007f189e38720579f'] Type: i3.large): Init nodes
(...)
< t:2020-06-19 18:52:09,915 f:cluster.py      l:2647 c:sdcm.cluster         p:INFO  > Cluster multi-cluster-kbraun-loader-set-cd32b68f (AMI: ['ami-0ce9dc296997751b7'] Type: c4.large): Init nodes
```

Here I have 3 completely independent clusters. The timestamps suggest that they weren't really setup in parallel. In this particular example not only the AWS instances can be setup in parallel, but the 2 Scylla nodes as well, since they are in two separate clusters (`db_cluster` and `cs_db_cluster`).

Another example from this test: I'm bootstrapping a new node in the middle of the test. This requires to setup an AWS instance, but the instance could already be waiting for me from the beginning of the test, and I would just start Scylla on it.

A lot can be improved. Looking at the above `output.log`, we see that the test started at `18:48:34` and ended at `19:42:32`, which means that the test has been running for ~56 minutes. It took 16 minutes to fetch the CDC log at the end of the test and I'm going to remove this part (also, this only happens if the test fails), so without it or in a successful test run we would have ~30 minutes. However, if we did as much in parallel as we could, it could work as follows:
1. Set up 5 AWS instances: ~2 minutes
2. Start a 3-node Scylla cluster + independent 1-node (oracle) cluster + install necessary stuff on loader node: ~3 minutes (assuming we give 1 node 1 minute to bootstrap)
3. Run C-S for 15 minutes. In the middle bootstrap a 4th node to the 3-node cluster, but that's included in those 15 minutes.
4. Some sleeps here and there, about 2 minutes in total.

In total: ~(2 + 3 + 15 + 2) = ~22 minutes.

This is 8/30 = 26% redunction in test running time. It might not seem much for a single run, but if I wanted to do many runs of this test in a day, it gives a significant difference (maybe between catching and not catching a bug this day).

Well, I could be running these test instances in parallel, so I'm not convinced if it's the best argument.

But here's a different argument, and I think it convinces me the most: say I'm developing a test, or I'm developing a feature and running some SCT tests while developing the feature to do some checks. Then, the shorter the test runs, the more efficient my work is. I'm trying to do work in parallel while the test is running, but sometimes I need its result to decide on what to do next; I could context-switch to another task, but that is not always so easy and breaks your flow. Waiting for tests to finish is sometimes the most annoying part of the development cycle. In fact, this is the reason that I opened this issue.
